common/setup.c

```diff
@@ -44,12 +45,78 @@
 #include <drivers/serial.h>
 #include <slab.h>
 
-bool opt_debug;
+bool opt_debug = false;
+bool_cmd("debug", opt_debug);
 
 io_port_t com_ports[2];
 
 const char *kernel_cmdline;
 
+static __text_init int parse_bool(const char *s) {
+    if (!strcmp("no", s) || !strcmp("off", s) || !strcmp("false", s) ||
+        !strcmp("disable", s) || !strcmp("0", s))
+        return 0;
+
+    if (!strcmp("yes", s) || !strcmp("on", s) || !strcmp("true", s) ||
+        !strcmp("enable", s) || !strcmp("1", s))
+        return 1;
+
+    return -1;
+}
+
+void __text_init cmdline_parse(const char *cmdline) {
+    static __bss_init char opt[PAGE_SIZE];
+    char *optval, *optkey, *q;
+    const char *p = cmdline;
+    struct ktf_param *param;
+
+    if (cmdline == NULL)
+        return;
+
+    for (;;) {
+        p = string_trim_whitspace(p);
+
+        if (iseostr(*p))
+            break;
+
+        q = optkey = opt;
+        while ((!isspace(*p)) && (!iseostr(*p))) {
+            ASSERT((q - opt) < (long) (PAGE_SIZE - 1));
```

Comment:
nit: `ASSERT((q - opt) < (long) (sizeof(opt) - 1));`

Does it have to be `ASSERT((q - opt) < (long) (sizeof(opt) - 1));` or is `ASSERT((q - opt) < sizeof(opt));` enough?

Reply:
sizeof(opt) should be fine as well!